packages/story-editor/src/app/taxonomy/taxonomyProvider.js

```diff
@@ -145,50 +146,46 @@ function TaxonomyProvider(props) {
   );
 
   const addSearchResultsToCache = useCallback(
-    async (
-      taxonomy,
-      {
-        name,
-        // This is the per_page value Gutenberg is using
-        perPage = 20,
-      },
-      addNameToSelection = false
-    ) => {
+    async (taxonomy, args, addNameToSelection = false) => {
       let response = [];
       const termsEndpoint = taxonomy['_links']?.['wp:items']?.[0]?.href;
       if (!termsEndpoint) {
         return;
       }
       try {
-        response = await getTaxonomyTerm(termsEndpoint, {
-          search: name,
-          per_page: perPage,
-        });
+        response = await getTaxonomyTerm(termsEndpoint, args);
       } catch (e) {
-        // Do we wanna do anything here?
+        // TODO #9033
       }
 
       // Avoid update if we're not actually adding any terms here
       if (response.length < 1) {
-        return;
+        // eslint-disable-next-line consistent-return
+        return response;
       }
 
       // Format results to fit in our { [taxonomy]: { [slug]: term } } map
       const termResults = {
         [taxonomy.restBase]: dictionaryOnKey(response, 'slug'),
       };
+
       setTermCache((cache) => mergeNestedDictionaries(cache, termResults));
 
-      if (addNameToSelection) {
-        const selectedTermSlug = cleanForSlug(name);
+      if (addNameToSelection && args.search) {
+        const selectedTermSlug = cleanForSlug(args.search);
         const selectedTerm = response.find(
           (term) => term.slug === selectedTermSlug
         );
 
         if (selectedTerm) {
           addTermToSelection(taxonomy, selectedTerm);
         }
+      } else if (args.search) {
+        setFlatSearchResults(response);
```

Comment:
So I think this approach won't be taxonomy agnostic. I would just grab the results from this function in the `FlatTermSelector` and store them in state there, then they will be taxonomy agnostic. Basically just do all this state keeping in that component vs in the provider here. shouldn't be too bad of an update since it shouldn't really change how you're using `flatSearchResults` will just be stored in a different place